libs/designsystem/src/lib/components/radio/radio-group/radio-group.component.spec.ts

```diff
@@ -697,4 +704,540 @@ describe('RadioGroupComponent', () => {
+            newSelectedIndex = defaultSelectedIndex + 1;
+            await setSelectedOnHostComponent(items[newSelectedIndex]);
+            // Wait for radio checked attribute to be updated;
+            await TestHelper.whenTrue(() => radioChecked(newSelectedIndex));
+
+            expect(radioChecked(0)).toBeFalse();
+            expect(radioChecked(1)).toBeFalse();
+            expect(radioChecked(2)).toBeTrue();
+          });
+
+          it('should not emit change event when the bound ngModel is updated', async () => {
+            const onChangeSpy = spyOn(spectator.component.valueChange, 'emit');
+            await setSelectedOnHostComponent(items[2]);
+            expect(onChangeSpy).not.toHaveBeenCalled();
+          });
+        });
+      });
+
+      describe('error state when ngModel is required', () => {
+        beforeEach(async () => {
+          spectator = createHost(
+            `<kirby-radio-group [items]="items" [(ngModel)]="selected" required>
+            </kirby-radio-group>`,
+            {
+              hostProps: {
+                items: items,
+                selected: null,
+              },
+            }
+          );
+        });
+
+        describe('when ngModel is not null', () => {
+          beforeEach(async () => {
+            await setSelectedOnHostComponent(items[defaultSelectedIndex]);
+          });
+
+          describe('and ngModel is touched', async () => {
+            beforeEach(async () => {
+              spectator.component._onRadioBlur();
+              await TestHelper.waitForTimeout();
+              spectator.detectChanges();
+
+              expect(spectator.element.classList).toContain('ng-touched');
+            });
+
+            it('should not be in error state', async () => {
+              expect(spectator.element.classList).not.toContain('ng-invalid');
+            });
+          });
+
+          describe('and ngModul is untouched', async () => {
+            it('should not be in error state', async () => {
+              expect(spectator.element.classList).not.toContain('ng-touched');
+              expect(spectator.element.classList).not.toContain('ng-invalid');
+            });
+          });
+        });
+
+        describe('When ngModel is null', () => {
+          beforeEach(async () => {
+            await setSelectedOnHostComponent(null);
+          });
+
+          describe('and ngModel is touched', () => {
+            beforeEach(async () => {
+              spectator.component._onRadioBlur();
+              await TestHelper.waitForTimeout();
+              spectator.detectChanges();
+
+              expect(spectator.element.classList).toContain('ng-touched');
+            });
+
+            it('should be in error state', async () => {
+              expect(spectator.element.classList).toContain('ng-touched');
+              expect(spectator.element.classList).toContain('ng-invalid');
```

Comment:
Consider expecting the rendering of the radios instead:
```suggestion
        //TODO loop radioElements, get icon for each and assert on rendered style:
         expect(radioIcon).toHaveComputedStyle({
        'border-width': '1px',
        'border-color': getColor('danger'),
      });
```